From 44c5e28b2bd4679056a0cc36e4ed8901ba02e2d9 Mon Sep 17 00:00:00 2001
From: Rogerio Guerra Borin <rogerio.borin@toradex.com>
Date: Fri, 8 May 2026 20:23:48 -0300
Subject: [PATCH] apalis-imx6: add fastboot env variables (upstream)

Add some U-Boot environment variables to simplify flashing via Fastboot:

- fastboot_partition_alias_all,
  fastboot_raw_partition_spl,
  fastboot_raw_partition_bootloader:
  Variables utilized by UUU flashing scripts; they allow users to
  specify the names "all", "spl" and "bootloader" in the flash command.
  For the spl and bootloader partitions, a raw definition is used to
  deal with the needed non-zero offset.
- emmc_dev, emmc_ack:
  These are also utilized by standard/built-in UUU flashing scripts;
  the former indicates the eMMC device to be used with flashing and
  the latter indicates if the device requires an ACK during boot (info
  passed to the "mmc partconf" command when configuring the device).
- fastboot_default_buf_addr, fastboot_default_buf_size:
  Additional variables indicating the location and size of the default
  Fastboot buffer; this helps avoid the need for users to hard-code
  these pieces of information in their flashing scripts.

Upstream-Status: Inappropriate [Toradex specific]
Signed-off-by: Rogerio Guerra Borin <rogerio.borin@toradex.com>
---
 include/configs/apalis_imx6.h | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
index 0df3917e200..60762277f30 100644
--- a/include/configs/apalis_imx6.h
+++ b/include/configs/apalis_imx6.h
@@ -62,12 +62,32 @@
 	"ramdisk_addr_r=0x18400000\0" \
 	"scriptaddr=0x18280000\0"
 
+#ifdef CONFIG_FASTBOOT_FLASH
+# ifndef CONFIG_FASTBOOT_FLASH_MMC_DEV
+#  error CONFIG_FASTBOOT_FLASH_MMC_DEV must be defined
+# endif
+#define FASTBOOT_ENV_SETTINGS \
+	"fastboot_partition_alias_all=" \
+		__stringify(CONFIG_FASTBOOT_FLASH_MMC_DEV) ".0:0\0" \
+	"fastboot_raw_partition_spl=0x2 0x88 mmcpart 1\0" \
+	"fastboot_raw_partition_bootloader=0x8a 0x1f76 mmcpart 1\0" \
+	"fastboot_default_buf_addr=" \
+		__stringify(CONFIG_FASTBOOT_BUF_ADDR) "\0" \
+	"fastboot_default_buf_size=" \
+		__stringify(CONFIG_FASTBOOT_BUF_SIZE) "\0" \
+	"emmc_dev=" __stringify(CONFIG_FASTBOOT_FLASH_MMC_DEV) "\0" \
+	"emmc_ack=1\0"
+#else
+#define FASTBOOT_ENV_SETTINGS
+#endif
+
 #define CFG_EXTRA_ENV_SETTINGS \
 	BOOTENV \
 	"boot_script_dhcp=boot.scr\0" \
 	"console=ttymxc0\0" \
 	"fdt_board=eval\0" \
 	MEM_LAYOUT_ENV_SETTINGS \
+	FASTBOOT_ENV_SETTINGS \
 	UBOOT_UPDATE \
 	"setethupdate=if env exists ethaddr; then; else setenv ethaddr " \
 		"00:14:2d:00:00:00; fi; tftpboot ${loadaddr} " \
-- 
2.43.0

